test/codegen-generated-classes-test/src/test/java/software/amazon/awssdk/services/protocolrestjson/StringPayloadUnmarshallingTest.java

```diff
@@ -0,0 +1,180 @@
+import org.junit.jupiter.params.provider.MethodSource;
+import software.amazon.awssdk.auth.credentials.AwsBasicCredentials;
+import software.amazon.awssdk.auth.credentials.StaticCredentialsProvider;
+import software.amazon.awssdk.regions.Region;
+
+@WireMockTest
+public class StringPayloadUnmarshallingTest {
+    private static List<Arguments> testParameters() {
+        List<Arguments> testCases = new ArrayList<>();
+        for (ClientType clientType : ClientType.values()) {
+            for (Protocol protocol : Protocol.values()) {
+                for (StringLocation value : StringLocation.values()) {
+                    for (ContentLength contentLength : ContentLength.values()) {
+                        testCases.add(Arguments.arguments(clientType, protocol, value, contentLength));
+                    }
+                }
+            }
+        }
+        return testCases;
+    }
+
+    private enum ClientType {
+        SYNC,
+        ASYNC
+    }
+
+    private enum Protocol {
+        JSON
+        // TODO - add support for XML
+    }
+
+    private enum StringLocation {
+        PAYLOAD,
+        FIELD
+    }
+
+    private enum ContentLength {
+        ZERO,
+        CHUNKED_ZERO
+    }
+
+    @ParameterizedTest
+    @MethodSource("testParameters")
+    public void missingStringPayload_unmarshalledCorrectly(ClientType clientType,
+                                                           Protocol protocol,
+                                                           StringLocation stringLoc,
+                                                           ContentLength contentLength,
+                                                           WireMockRuntimeInfo wm) {
+        if (contentLength == ContentLength.ZERO) {
+            stubFor(post(anyUrl()).willReturn(aResponse().withStatus(200).withHeader("Content-Length", "0").withBody("")));
+        } else if (contentLength == ContentLength.CHUNKED_ZERO) {
+            stubFor(post(anyUrl()).willReturn(aResponse().withStatus(200).withBody("")));
+        }
+
+        String serviceResult = callService(wm, clientType, protocol, stringLoc);
+
+        if (stringLoc == StringLocation.PAYLOAD) {
+            assertThat(serviceResult).isNotNull().isEqualTo("");
+        } else if (stringLoc == StringLocation.FIELD) {
+            assertThat(serviceResult).isNull();
+        }
+    }
+
+    @ParameterizedTest
+    @MethodSource("testParameters")
+    public void presentStringPayload_unmarshalledCorrectly(ClientType clientType,
+                                                           Protocol protocol,
+                                                           StringLocation stringLoc,
+                                                           ContentLength contentLength,
+                                                           WireMockRuntimeInfo wm) {
+        String responsePayload = presentStringResponse(protocol, stringLoc);
+
+        if (contentLength == ContentLength.ZERO) {
```

Comment:
What does `ContentLength` mean in the context of this test? It seems to be we're tacking new meanings (cl vs no cl) onto `ZERO` and `CHUNKED_ZERO` respectively. Is there a way we can express it so it makes sense?